core/src/test/scala/unit/kafka/server/BaseRequestTest.scala

```diff
@@ -99,10 +110,21 @@ abstract class BaseRequestTest extends KafkaServerTestHarness {
       byteBuffer.array()
     }
 
-    val response = requestAndReceive(serializedBytes)
+    val response = requestAndReceive(socket, serializedBytes)
 
     val responseBuffer = ByteBuffer.wrap(response)
     ResponseHeader.parse(responseBuffer) // Parse the header to ensure its valid and move the buffer forward
     responseBuffer
   }
+
+  def validateApiVersionsResponse(apiVersionsResponse: ApiVersionsResponse) {
```

Comment:
How about we put this in the companion object of `ApiVersionsTest`? It's nicer if we can keep the base request class free of specific request validation logic.